hasher.go

```diff
@@ -71,16 +74,70 @@ func (n *Hasher) NamespaceSize() namespace.IDSize {
 
 func NewNmtHasher(baseHasher hash.Hash, nidLen namespace.IDSize, ignoreMaxNamespace bool) *Hasher {
 	return &Hasher{
-		Hash:             baseHasher,
+		baseHasher:       baseHasher,
 		NamespaceLen:     nidLen,
 		ignoreMaxNs:      ignoreMaxNamespace,
 		precomputedMaxNs: bytes.Repeat([]byte{0xFF}, int(nidLen)),
 	}
 }
 
+// Size returns the number of bytes Sum will return.
+func (n *Hasher) Size() int {
+	return n.baseHasher.Size() + int(n.NamespaceLen)*2
+}
+
+// Write writes the namespaced data to be hashed.
+//
+// Requires data of fixed size to match leaf or inner NMT nodes.
+// Only a single write is allowed.
+func (n *Hasher) Write(data []byte) (int, error) {
```

Comment:
Does this implement another interface? If so, can you document? If it doesn't implement an interface then I'd say just don't allow more than one call to `Write` by construction.

Reply:
It implements `hash.Hash`, which implements `io.Writer`.